### ISSUE TITLE
Double click event not being fired

### ISSUE THREAD

**Issue by chipit24** [clientIO/joint#1263]
**Note:** Jump down to the **Update** below, as the case without multiple `toFront` calls is expected behaviour for double clicks.
_____
I found that when I click multiple times on an element, the `cell:pointerclick` event is fired every single time, but the `cell:pointerdblclick` event is not always fired, and if I click fast enough, it doesn't seem to fire at all. This is causing issues for me.

Here is a GIF showing what's happening:
![double-click-not-always-fired](https://user-images.githubusercontent.com/7561061/76327170-08da7980-62c0-11ea-8282-6a53f9b7730c.gif)

I created the above in CodeSandbox: https://codesandbox.io/s/cocky-surf-lbs76.

The code is below:
```js
import { dia, shapes } from "jointjs";

const graph = new dia.Graph();
const paper = new dia.Paper({
  el: document.getElementById("app"),
  model: graph
});

paper.on("cell:pointerdblclick", () => console.log("Double click!"));
paper.on("cell:pointerclick", () => console.log("Click!"));

const rectangle = new shapes.standard.Rectangle();
rectangle.resize(100, 100);
rectangle.position(50, 10);
rectangle.addTo(graph);
```

I played around with it, and it appears this behaviour gets worse (the double click event is fired less often) if I add more shapes and links between them. 

This is testing with JointJS 3.1.1, in Chrome 80, FF 74, and Safari 13.

**Update:**
I created another CodeSandbox: https://codesandbox.io/s/friendly-mendeleev-6d8tp.
If I call `toFront` multiple times, e.g.:
```js
paper.on("cell:pointerdown", () => {
  rectangle.toFront({ deep: true });
  rectangle2.toFront({ deep: true });
  link.toFront({ deep: true });
});
```
Then the double click event isn't even fired once for any of the elements (in Safari 13.0.4).

**Comment by kumilingus** [comment]
Isn't a standard browser behaviour?
<img width="545" alt="image" src="https://user-images.githubusercontent.com/3967880/76377463-edd23d00-6385-11ea-95f7-78bcf03ef1d0.png">

The other issue with `toFront()` could be related to https://github.com/clientIO/joint/issues/585. Does it happen in any other browser for you?

**Comment by chipit24** [comment]
I tried to distill this down to the most basic example, but you are right, the case without the multiple `toFront` calls is expected behaviour for double clicks.

I saw https://github.com/clientIO/joint/issues/585 but it didn't provide much help as this appears to be working well for me in Chrome. It's also working well for me in Firefox; only in Safari does it misbehave. The strange thing is, it's not working for my collegue on his machine in Chrome, despite also running the latest Chrome version (80).
